qualang_tools/optimal_weights/StateDiscriminator.py

```diff
@@ -0,0 +1,256 @@
+import numpy as np
+from pandas import DataFrame
+from qm.qua import *
+import os
+import matplotlib.pyplot as plt
+from sklearn import mixture
+from scipy import signal
+
+from TimeDiffCalibrator import TimeDiffCalibrator
+
+
+class StateDiscriminator:
+    """
+    The state discriminator is a class that generates optimized measure procedure for state discrimination
+    of a multi-level qubit.
+    .. note:
+        Currently only 3-states discrimination is supported. The setup assumed here includes IQ mixer both in the up-
```

Comment:
3-states?

Reply:
I don't understand your comment

Reply:
I think this discriminates between two states, am I'm wrong?

Reply:
I mean, it can be used for both two and three states, but the note says "Only 3 states"